py/server/deephaven/table.py

```diff
@@ -462,40 +503,35 @@ def size(self) -> int:
         """The current number of rows in the table."""
         return self.j_table.size()
 
-    @property
+    @cached_property
     def is_refreshing(self) -> bool:
         """Whether this table is refreshing."""
-        if self._is_refreshing is None:
-            self._is_refreshing = self.j_table.isRefreshing()
-        return self._is_refreshing
+        return self.j_table.isRefreshing()
 
-    @property
+    @cached_property
     def is_blink(self) -> bool:
         """Whether this table is a blink table."""
         return _JBlinkTableTools.isBlink(self.j_table)
 
-    @property
+    @cached_property
     def update_graph(self) -> UpdateGraph:
         """The update graph of the table."""
-        if self._update_graph is None:
-            self._update_graph = UpdateGraph(self.j_table.getUpdateGraph())
-        return self._update_graph
+        return UpdateGraph(self.j_table.getUpdateGraph())
 
-    @property
+    @cached_property
     def is_flat(self) -> bool:
         """Whether this table is guaranteed to be flat, i.e. its row set will be from 0 to number of rows - 1."""
-        if self._is_flat is None:
-            self._is_flat = self.j_table.isFlat()
-        return self._is_flat
+        return self.j_table.isFlat()
```

Comment:
`isFlat` is technically mutable, as is `isRefreshing`. That said, a Python wrapper should never see a table that hasn't had these set in their final form, so I don't know that it actually matters.

`isBlink` is effectively immutable, on the other hand; that is, once you look at it, it can't be changed (like anything else stored in attributes). `updateGraph` is `final`, also OK.

Reply:
If there is a possibility for mutation, I would remove the caching because we'll never remember to make a change here if we do get to a case where mutation happens.  I also doubt that these get called enough to make a significant performance difference.

Reply:
I'm convinced the caching is fine. But I also agree that we are likely not calling these anywhere performance critical.